enzy_htp/structure/structure_region/capping.py

```diff
@@ -50,124 +62,222 @@ def capping_with_residue_terminals(raw_region,
+    def CAP_BOND_DISTANCE(self) -> Dict[str, float]:
+        """Each specialization must have a charge map.""" 
+        pass
+
     @property
+    @abstractmethod
     def cap_type(self) -> str:
-        """hard coded cap type"""
-        _LOGGER.error("using abstract method.")
-        raise AttributeError
+        """What type of cap is this? Hardcoded value specific to each specialization."""
+        pass
+
+
+    @property
+    @abstractmethod
+    def net_charge(self) -> int:
+        """The int() net charge of the ResidueCap()."""
+        pass
+
+    @property
+    def idx(self) -> int:
+        """Gets the index of the link_residue."""
+        return self.link_residue.idx
 
     @property
-    def cap_bond_vector(self) -> np.array:
-        """return a vector that defines the direction of the capping bond"""
-        result = np.array(self.socket_atom.coord) - np.array(self.link_atom.coord)
-        result /= np.linalg.norm(result)
-        return result
+    def chain(self) -> Chain:
+        """Gets the chain name of the link_residue."""
+        return self.link_residue.parent
 
-    def init_fixed_atomic_charges(self) -> str:
+    @property
+    def name(self) -> str:
+        """Gets the residue name of the link_residue."""
+        return self.link_residue.name
+    
+    @property
+    def term_type(self) -> str:
+        """What type of capping is this? n-term, c-term, etc."""
+        return self._terminal_type
+
+    @term_type.setter
+    def term_type(self, val_in:str) -> None:
+        """Setter for the terminal type of the ResidueCap(). Does not perform validation."""
+        self._terminal_type = val_in
+
+    def init_fixed_atomic_charges(self) -> None:
         """init hard coded fixed atomic charges.
         Used when init_charge and fixed charge stragety is used."""
         for atom in self.atoms:
             atom.charge = self.FIXED_CHARGE_MAP[atom.name]
 
     def is_residue_cap(self) -> bool:
-        """hard coded yes"""
+        """Does this class cap a Residue? Always True."""
         return True
+
+    def is_nterm_cap(self) -> bool:
+        """Does this ResidueCap attach to the n-terminal end of an amino acid Residue?"""
+        return self.term_type == "nterm"
+
+    def is_cterm_cap(self) -> bool:
+        """Does this ResidueCap attach to the c-terminal end of an amino acid Residue?"""
+        return self.term_type == "cterm"
+
+    def __str__(self) -> str:
+        """str() rep of the ResidueCap() that shows its chemical type as well as what Atom()'s it is linked to."""
+        return f"{self.cap_type}Cap(link_atom={self.link_atom.name},link_residue={self.link_residue.key_str})"
+
+
+class HCap(ResidueCap):
+    """A ResidueCap() which uses a -H (hydrogen) as a cap. Adds a total of 1 atom1, with the 
     
-    def restore_default_coord(self):
-        for atom in self.atoms:
-            atom.coord = self.DEFAULT_COORDS[atom.name]
-
-    def apply_to_geom(self, geom: Structure) -> ResidueCap:
```

Comment:
StructureRegion use this function for atoms_from_geom and thus for single_point to deal with ensemble input. Please restore it.

Reply:
Please make `test/structure/structure_region/test_api.py::test_atoms_from_geom()` work

Reply:
Yes and yes

Reply:
Fixed! Pushed in commit